asset/asset.go

```diff
@@ -692,6 +728,11 @@ type Asset struct {
 	// together across distinct asset IDs, allowing further issuance of the
 	// asset to be made possible.
 	GroupKey *GroupKey
+
+	// XOnlyGroupKey is the tweaked public key that is used to associate
+	// assets together across distinct asset IDs, allowing further issuance
+	// of the asset to be made possible.
+	XOnlyGroupKey SchnorrKey
```

Comment:
Similar here as well: for a logical asset we want to know this information, but it can be obtained via another source.

Reply:
So maybe we actually want to modify the `AssetGroup` struct to include the tapscript root for the group key? For it to maintain the same definition of group key + info needed to derive it.

Reply:
Wouldn't that just live in the initial genesis proof? At that point you've already verified the derivation. 

I do agree that we need to store the group key derivation explicitly, so this way we can refer back to it (verify the inclusion proof for the tapscript leaf, etc) when it comes to further minting.

Reply:
> So maybe we actually want to modify the AssetGroup struct to include the tapscript root for the group key?

Or in something like `GroupDerivationInfo` or w/e which is read when we need to go verify subsequent minting events?